dangerzone/gui/main_window.py

```diff
@@ -448,49 +487,83 @@ def __init__(self, dangerzone: DangerzoneGui) -> None:
+                    log.error(f"{runtime_name} is not running")
                     state = "not_running"
+                    error = stderr.decode()
                 else:
                     # Always try installing the container
                     state = "install_container"
 
         # Update the state
-        self.state_change(state)
+        self.state_change(state, error)
 
-    def state_change(self, state: str) -> None:
+    def state_change(self, state: str, error: Optional[str] = None) -> None:
         if state == "not_installed":
-            self.label.setText(
-                "<strong>Dangerzone Requires Docker Desktop</strong><br><br><a href='https://www.docker.com/products/docker-desktop'>Download Docker Desktop</a>, install it, and open it."
-            )
+            if platform.system() == "Linux":
+                self.label.setText(
+                    (
+                        "<strong>Dangerzone requires Podman</strong><br><br>"
+                        "Install it and retry."
+                    )
+                )
+            else:
+                self.label.setText(
+                    (
+                        "<strong>Dangerzone requires Docker Desktop</strong><br><br>"
+                        "<a href='https://www.docker.com/products/docker-desktop'>Download Docker Desktop</a>"
+                        ", install it, and open it."
+                    )
+                )
             self.buttons.show()
         elif state == "not_running":
-            self.label.setText(
-                "<strong>Dangerzone Requires Docker Desktop</strong><br><br>Docker is installed but isn't running.<br><br>Open Docker and make sure it's running in the background."
-            )
+            if platform.system() == "Linux":
+                # "not_running" here means that the `podman image ls` command failed.
+                message = (
+                    "<strong>Dangerzone requires Podman</strong><br><br>"
+                    "Podman is installed but cannot run properly. See errors below"
+                )
+                if error:
+                    self.traceback.set_content(error)
+
+                self.label.setText(message)
+
+            else:
+                self.label.setText(
+                    (
+                        "<strong>Dangerzone requires Docker Desktop</strong><br><br>"
+                        "Docker is installed but isn't running.<br><br>"
+                        "Open Docker and make sure it's running in the background."
+                    )
+                )
             self.buttons.show()
         else:
             self.label.setText(
-                "Installing the Dangerzone container image.<br><br>This might take a few minutes..."
+                (
+                    "Installing the Dangerzone container image.<br><br>"
+                    "This might take a few minutes..."
+                )
             )
             self.buttons.hide()
```

Comment:
Besides hiding the buttons, we may need to hide the error box (`self.error_text`) as well. Else, if a user fixes a broken Podman installation, the error box will remain while the container image gets installed.

Piping the installation progress of the container image would be the ideal solution here, but it's an overkill for this feature.

Reply:
Good catch, I've made the change.